lib/analyzer/applications/runtime-common.ts

```diff
@@ -1,23 +1,82 @@
 import * as path from "path";
+import { parsePkgJson } from "snyk-nodejs-lockfile-parser";
 import { ApplicationFilesFact } from "../../facts";
 import {
   AppDepsScanResultWithoutTarget,
   ApplicationFileInfo,
   FilePathToContent,
 } from "./types";
 
-export function getAppFilesRootDir(
-  filePaths: string[],
-): [string, ApplicationFileInfo[]] {
+interface AppFileMetadataExtractor {
+  manifestFiles: string[];
+  metadataExtractor: (fileContent: string) => Record<string, string>;
+}
+
+export const filesMetadataExtractorPerLanguage: Record<
+  string,
+  AppFileMetadataExtractor
+> = {
+  node: {
+    manifestFiles: ["package.json", "package-lock.json", "yarn.lock"],
+    metadataExtractor: (fileContent: string) => {
+      const pkgJson = parsePkgJson(fileContent);
+      const repoUrl = (pkgJson as any).repository?.url;
+      return {
+        moduleName: pkgJson.name,
+        repoUrl,
+      };
+    },
+  },
+};
+
+export function getAppFileInfos(
+  filePathToContent: FilePathToContent,
+  rootDir: string,
+  manifestMetadataExtractor?: AppFileMetadataExtractor,
+): ApplicationFileInfo[] {
   const appFiles: ApplicationFileInfo[] = [];
-  const splitPaths: string[][] = [];
 
+  const filePaths = Object.keys(filePathToContent);
   if (!filePaths.length) {
-    return [path.sep, appFiles];
+    return appFiles;
   }
 
   for (const filePath of filePaths) {
-    appFiles.push({ path: filePath });
+    const appFile: ApplicationFileInfo = { path: filePath };
+    if (manifestMetadataExtractor) {
+      const { manifestFiles, metadataExtractor } = manifestMetadataExtractor;
+      if (
+        manifestFiles.length &&
+        manifestFiles.some((mf) => filePath.endsWith("/" + mf))
```

Comment:
Please use path.base to extract the file name, otherwise it will fail if the manifest is in the root dir